nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -1010,8 +1010,15 @@ public DocumentRef getParentDocumentRef(DocumentRef docRef) {
 
     @Override
     public DocumentModel getParentDocument(DocumentRef docRef) {
-        Document doc = resolveReference(docRef);
-        Document parentDoc = doc.getParent();
+        Document parentDoc;
+        if (docRef != null && docRef.type() == DocumentRef.PATH) {
+            String targetPath = ((String) docRef.reference());
```

Comment:
Maybe simply `docPath` instead of `targetPath`, as we have `docRef`, and `doc`.

Reply:
Also remove extra parentheses.